### PR TITLE
Added a feature which will allow the user to control space b/w value and unit

### LINKED ISSUE #7455: Space between value and unit
Currently, ``Angle.to_string`` doesn't include a space between the value and unit:

```python
In [30]: from astropy.coordinates import Angle

In [31]: a = Angle(3, 'deg')

In [32]: a.to_string(unit='mas')
Out[32]: '1.08e+07mas'
```

I think there are cases where it would make sense to allow a space to be included, so this is a feature request to add a boolean keyword argument to optionally add a space.

Note that Quantity does include a space by default so maybe actually we should just change the default and not add an option?

```python
In [17]: str(3 * u.mas)
Out[17]: '3.0 mas'
```

### PR DESCRIPTION
**Pull request by Telomelonia**
### Description

I added a boolean feature `oldstyle` in `to_string()`, giving flexibility to the user to have a space between value and unit.
By default, space should be given in between whenever its possible.
**NOTE**-For some units like 34'20'' or 11<sup>h</sup>22<sup>m</sup> no space should be added whether the feature is set to `True` or `False`.
```
oldstyle : bool, optional
            If `False`, include a space between value and unit. If `True`,
            no space is included. Doesn't work with 'latex' or 'unicode'
            format. By default a space is generated.
```

### TODOs
- [x] Adding the feature as a boolean keyword
- [x] Updating the Docstring
- [x] Add a changelog
- [x] Update Testing

This pull request is to address ...

Fixes #7455 

### Testing
image
image

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
The tests need updating. A new test to use the alternate option is also necessary.

Also a change log is necessary (as you have noted in your checklist).

Thanks!

File: astropy/coordinates/angles.py
Comment (mhvk) [inline astropy/coordinates/angles.py:212]: This name is not very informative. `Quantity.to_string`, which `Angle` as a `Quantity` subclass should ideally follow, has `subfmt`, which I guess one could use instead.

**Review by mhvk — CHANGES_REQUESTED** [review]
Thanks for the PR, but I wonder why this is needed? Most journals insist that a space is used between a number and the unit. Also, while for simple units this might make sense, it does not generalize to composite units (which have spaces in between them).

Overall, my tendency would be to leave this for people to sort for themselves if they want it; generally `.to_string(...).replace(' ', '')` will be all that's needed.

So, not in favour; marking as "request changes" since if we were to go for adding this option, it definitely should have a different name.

**Comment by mhvk** [comment]
Oops, I'm so sorry, I hadn't even realized that the issue here is to *fix* that angles do not have spaces between the number and the unit. Very silly of me; I should have read the issue.

Now having done so, I agree with the comments there that the default should be to have the space, with the only exception being when one of the special symbols is used. My tendency would be not to even have an option for this.

Anyway, now *in favour* as long as we change the default!

File: astropy/coordinates/angles.py
Comment (mhvk) [inline astropy/coordinates/angles.py:334]: I wonder if the whole `else` clause could simply leave it to `Quantity.to_string()` via
```
return super().to_string(unit=unit, precision=precision, format=format)
```

File: docs/changes/coordinates/14231.feature.rst
Comment (mhvk) [inline docs/changes/coordinates/14231.feature.rst:1]: Maybe for once decisions have been made, but I think this should be an `api` that states that spaces are now inserted before units in `Angle.to_string()`.

**Review by mhvk — COMMENTED** [review]
A few more thoughts.

**Comment by Telomelonia** [comment]
@pllim 
I have a problem regarding the pre-commit.ci check. Apart from that all is good to go

**Comment by pllim** [comment]
pre-commit.ci autofix

File: astropy/coordinates/angles.py
Comment (pllim) [inline astropy/coordinates/angles.py:278]:
```suggestion
            no space is included. This option does not work with 'latex' or 'unicode'
```

**Review by pllim — COMMENTED** [review]
A test need to be added with `oldstyle=True` to ensure it works as intended.

Please renamed the change log fragment file from `14231.feature.rst` to `14231.api.rst` as suggested by reviewer. Change log text also need to elaborate what this ``oldstyle`` actually does and why it is needed now.

Needs to squash all commits into one before this is ready for merge too. To do that, please remember to pull in pre-commit autofix commit from GitHub back to your local checkout first. Feel free to ask if you are unsure how to do this part.

**Comment by eerovaher** [comment]
> I have a problem regarding the pre-commit.ci check. Apart from that all is good to go

`pre-commit` can fix most (if not all) issues it highlights, see https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit

**Comment by Telomelonia** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
Re: https://github.com/astropy/astropy/pull/14231#issuecomment-1422651982

@mhvk , I think that is a discussion for astropy-dev mailing list and/or dev telecon. I am going to turn this PR into draft until this discussion is resolved.

**Comment by mhvk** [comment]
Let's just ping the coordinate team - @eerovaher, @StuartLittlefair, @adrn, @eteq - on whether to make a space between a number and the unit the standard case (as it is for `Quantity`), or whether it is important to allow users to keep the "old style" where, for `Angle` only, the space was omitted (likely by mistake). I feel this is a quite clear case of only one correct way, and that effectively this is a bug fix, with little or no impact to anyone since it affects only how angles are shown in string form. So, I'm in favour of just changing, without an option to get the old behavor back (which, after all is just a `f"{angle.value}{angle.unit}"` away, probably easier to discover than a keyword argument!).

**Comment by StuartLittlefair** [comment]
Just to sat that in response to @mhvk's ping, I have no strong opinion on this. The only real issue I can anticipate is if third-parties had based their unit tests on the string output from angles, in which case I don't have much sympathy...

**Comment by Telomelonia** [comment]
I thinking of making another PR to address this issue as just a bug where a space is required in the result in between value and unit.

**Comment by mhvk** [comment]
@Telomelonia - it might indeed be good to have 2 PRs to compare, though it also is fine to work with this one, Thanks for persisting!!!

**Pull request closed without merging**
